src/arrays.js

```diff
@@ -2,41 +2,86 @@
 // These functions only need to work with arrays.
 // Do NOT use the built in array methods to solve these. forEach, map, reduce, filter, includes, etc.
 // You CAN use concat, push, pop, etc. but do not use the exact method that you are replicating
-// You can use the functions that you have already written to help solve the other problems
+// You can use the functions that you have already written for the other problems
 
 const each = (elements, cb) => {
   // Iterates over a list of elements, yielding each in turn to the `cb` function.
   // This only needs to work with arrays.
   // based off http://underscorejs.org/#each
+  for (let i = 0; i < elements.length; i++) {
+    cb(elements[i], i);
+  }
 };
-
+// each([1, 2, 3]), (item, index) =>{
+  // console.log(`${item} @ ${index}`)
 const map = (elements, cb) => {
   // Produces a new array of values by mapping each value in list through a transformation function (iteratee).
   // Return the new array.
+  const newArray = [];
+  each(elements, (item, index) => {
+    newArray.push(cb(item, index));
+  });
+  return newArray;
 };
 
 const reduce = (elements, cb, memo = elements.shift()) => {
   // Combine all elements into a single value going from left to right.
   // Elements will be passed one by one into `cb`.
   // `memo` is the starting value.  If `memo` is undefined then make `elements[0]` the initial value.
+  let considerFirst = true;
+  if (arguments.length < 3) {
+    memo = elements[0];
+    considerFirst = false;
+  }
+  each(elements, (element, index) => {
+    if (index > 0 || considerFirst) {
+      memo = cb(memo, element, index, elements);
```

Comment:
The callback function doesn't take accept the entire `elements` array in its function signature, so no need to pass it to `cb`.